### PR TITLE
Linting dicts.

### LINKED ISSUE #323: Refactor OrderedDict to dict
Python's `OrderedDict` class is known to be slower than the plain `dict`. Since 3.6 in CPython, `dict` objects have been ordered as well. This was made an API guarantee in 3.7 of the Python standard. However, within `flow` I do not believe we give an API guarantee of `dict` orderings anywhere. We do need the ordering as tests are currently written, but since tests aren't an API promise with the recent drop of 3.5 support, we can use the fact that in 3.6 we have the implementation order `dict` objects without needing the 3.7 standard for testing purposes.

Other opinions on this are welcome.

### PR DESCRIPTION
**Pull request by bdice**
## Description
Refactored dict usage, removed OrderedDict, and simplified dict comprehensions. Resolves #323.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: flow/project.py [hunk 1]
```diff
@@ -851,11 +849,9 @@ def op2(job):
...
-        # An OrderedDict is not necessary here, but is used to ensure
-        # consistent ordering of pretend submission output for templates.
-        self.operations = (
-            OrderedDict() if operations is None else OrderedDict(operations)
-        )
+        # Requires Python >=3.6: dict must be ordered to ensure consistent
+        # pretend submission output for templates.
+        self.operations = {} if operations is None else dict(operations)
         if operation_directives is None:
             self.operation_directives = dict()
```
Comment (b-butler) [inline flow/project.py:856]: We should probably be consistent here and use `{}`.

**Review by b-butler — CHANGES_REQUESTED** [review]
It may be worth grepping the repo for `dict()` and replacing with the literal `{}`. I caught one place where `dict()` is still used. Otherwise LGTM.

File: flow/project.py (hunk 1 above)
Comment (bdice) [reply flow/project.py:856]: Hmm. I grepped but must have missed this one.
```suggestion
            self.operation_directives = {}
```

**Comment by bdice** [comment]
I forgot to mention that some of the dict/{} changes may be encompassed in other PRs. I did a lot of work and then split it into separate PRs, so they wouldn’t conflict or have interdependency. I’d like to merge this as-is and I will take a second look to make sure no other refactorings were missed, after the other PRs I made are merged.

**Review by b-butler — APPROVED** [review]
LGTM

**Pull request merged**